pkg/tnf/handlers/imagepullpolicy/imagepullpolicy_test.go

```diff
@@ -0,0 +1,138 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package imagepullpolicy_test
+
+import (
+	"fmt"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/handlers/generic"
+	"github.com/test-network-function/test-network-function/pkg/tnf/identifier"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	testTimeoutDuration = time.Second * 5
+)
+
+var (
+	genericTestSchemaFile = path.Join("schemas", "generic-test.schema.json")
+	imagepullFilename     = "imagepullpolicy.json"
+	/* #nosec G101 */
+	expectedPassPattern  = "(?m)IfNotPresent"
+	expectedFailPattern  = "(?m)Always"
+	pathRelativeToRoot   = path.Join("..", "..", "..", "..")
+	pathToTestSchemaFile = path.Join(pathRelativeToRoot, genericTestSchemaFile)
+	testPodNameSpace     = "testnamespace"
+	testPodName          = "testPodname"
+	testContainerNum     = 0
+	testInputSuccess     = "IfNotPresent"
+	testInputFilure      = "Always"
+)
+
+func createTest() (*tnf.Tester, []reel.Handler, *gojsonschema.Result, error) {
+	values := make(map[string]interface{})
+	values["POD_NAMESPACE"] = testPodNameSpace
+	values["POD_NAME"] = testPodName
+	values["CONTAINER_NUM"] = testContainerNum
+	return generic.NewGenericFromMap(imagepullFilename, pathToTestSchemaFile, values)
+}
+
+func TestImagePullPolicy_Args(t *testing.T) {
+	test, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Nil(t, (*test).Args())
+}
+
+func TestImagePullPolicy_GetIdentifier(t *testing.T) {
+	test, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, identifier.ImagePullPolicyIdentifier, (*test).GetIdentifier())
+}
+
+func TestImagePullPolicy_ReelFirst(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	step := handler.ReelFirst()
+	expectedCommand := fmt.Sprintf("oc get pod %s -n %s -o json  | jq -r '.spec.containers[%d].imagePullPolicy'", testPodName, testPodNameSpace, testContainerNum)
+	assert.Equal(t, expectedCommand, step.Execute)
```

Comment:
what would happen if this policy is not defined ?

Reply:
it seems it's always there when you dump it. it is optional when you create the pod. we found conflicting references about what the default is when it's missing.